docs/considerations.md

```diff
@@ -0,0 +1,68 @@
+Performance Considerations
+==========================
+
+Link:
+https://jira.owncloud.com/browse/OCIS-1705
+
+## How can we model typical scenarios?
+
+Getting comparable statement about performance of EFSS cloud installations is a non trivial task.
+
+To solve that, the idea is to model typical scenarios for EFSS cloud installations. That will make it possible to compare clouds objectively in terms of performance and stability.
+
+There needs to be a standardized process to collect data that allow a statement. It needs to be repeatable and clearly specified.
+
+For that, ownCloud developed a tool called cdperf. It automates to run certain test setups for currently three different test candidates: ownCloud 10, oCIS and NextCloud.
+
+## Influencual Parameters
+
+First, it makes sense to think about all parameters that influence the performance of an EFSS system.
+
+### Hardware
+
+It is important to use capable hardware for the test. Capable means that enough resources on the computer are available for the test, so that during testing no hardware limits are reached. That is especially important if networking is involved.
+
+Tests that should be compared need to be run on the same hardware in the same environment and on the same operating system.
+
+### Machine Setup
+
+Leaving things like hardware and operating system aside, the setup, ie. a virtualization- or bare metal setup, is important. To make it easy to repeat tests with different clouds on the same system, cdperf is using docker to run the cloud installations and the test tool.
+
+By default, it uses the officially provided docker containers of the cloud flavours. The installations within might be an more or less optimal setup.
+
+### Runtime Configuration
+
+Especially for a distributed system like oCIS it is important how the runtime manages the services. Are they run as separate processes, system threads or go co-routines? That influences how many operations can be parallized and how many processor cores can be used at the same time.
+
+### Concurrent Access
+
+To model a real life scenario of EFSS it is important to mimic parallel access to the system. That means that different users access the cloud independently at the same time.
```

Comment:
To model a real life scenario of EFSS it is important to mimic parallel access to the system. That means that many virtual users access the cloud independently at the same time.